### PR TITLE
Modified letters generator route to be POST and updated tests

### PR DESCRIPTION
**Pull request by samcoforma**
## Summary

- Updated route and related RSpec test to accept POST requests
- Fixes bug for Benefit Summary downloads with options

## Related issue(s)
- [department-of-veterans-affairs/va.gov-team#58661](https://github.com/department-of-veterans-affairs/va.gov-team/issues/58661)
- Epic: https://github.com/department-of-veterans-affairs/va.gov-team/issues/issues/58642

## Testing done

- RSpec tests updated and passing

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by kjduensing — APPROVED** [review]
LGTM

File: app/controllers/v0/letters_generator_controller.rb [hunk 1]
```diff
@@ -31,9 +32,7 @@ def index
     def download
       permitted_params = params.permit(DOWNLOAD_PARAMS)
       letter_options =
-        permitted_params.to_h
-                        .select { |_, v| v == 'true' }
-                        .transform_values { |v| ActiveModel::Type::Boolean.new.cast(v) }
+        permitted_params.to_h.transform_keys { |k| k.camelize(:lower) }
```
Comment (kjduensing) [inline app/controllers/v0/letters_generator_controller.rb:35]: What changed to where we're guaranteed the param options are boolean `true/false`?

Comment (samcoforma) [reply app/controllers/v0/letters_generator_controller.rb:35]: @kjduensing after switching to POST, they don't come in as strings anymore, but true booleans!

Comment (kjduensing) [reply app/controllers/v0/letters_generator_controller.rb:35]: Oh interesting, thanks

**Review by kjduensing — DISMISSED** [review]
LGTM

File: app/controllers/v0/letters_generator_controller.rb [hunk 2]
```diff
@@ -10,17 +10,18 @@ class LettersGeneratorController < ApplicationController
...
-      adaptedHousing
-      chapter35Eligibility
-      deathResultOfDisability
-      survivorsAward
+      special_monthly_compensation
+      adapted_housing
+      chapter35_eligibility
+      death_result_of_disability
+      survivors_award
+      letters_generator
```
[the remark this answers is not in the document]
Comment (samcoforma) [reply app/controllers/v0/letters_generator_controller.rb:24]: @kjduensing This is due to wanting to minimize changes made to the front-end. The front-end calls this endpoint with snake case params, but the Lighthouse API wants camelCase. So converting them here. Thoughts?

**Comment by samcoforma** [comment]
@kjduensing I had to modify the vcr cassettes to make sure that false options were being passed to lighthouse. Could you re-approve this PR?

**Review by ryan-mcneil — APPROVED** [review]
👌

**Pull request merged**